### DISCUSSION TITLE
Demo: how to turn clipboard content into a mind map in 5 seconds using AI

### DISCUSSION THREAD

**Discussion by metacoma** [freeplane/freeplane#2091] [Show and tell]
Hey folks!

I simply select the area of data I’m interested in on the screen and get its representation as a mind map.

https://github.com/user-attachments/assets/eb943c35-4be8-4e70-bbe7-4a9e1cf0105f

The "full" video: https://youtu.be/tKSRnIMWuMc

P.S. I’m using a custom GRPC plugin for Freeplane that allows me to control the mind map. Unfortunately, I don’t have the time or expertise to [fix these java traces.](https://github.com/metacoma/freeplane_plugin_grpc/issues). They don’t occur often, but I would be grateful for any help.

**Comment by euu2021** [comment]
Interesting!

You can also create a clipboard monitor (like [this](https://docs.freeplane.org/scripting/Scripts_collection.html#monitor-clipboard)) so you don't even need to paste into Freeplane.

**Reply by metacoma** [reply]
Hey @euu2021, thank you!

Yes, I know about this script. In my case, I use the following workflow:

1. Use [clipnotify](https://github.com/cdown/clipnotify) to detect changes in buffers.
2. [Pass the clipboard data to ChatGPT.](https://github.com/metacoma/freeplane_plugin_grpc/blob/main/misc/chatgpt/send_request.py)
3. Read the answer from ChatGPT.
4.[Call the improtJson gRPC function in the Freeplane instance](https://github.com/metacoma/freeplane_plugin_grpc/blob/7e5bb18effcc59af09f94f906d9adcd49d09d642/grpc/shell/import_json.sh#L4)

All neccessary stuff can be found here https://github.com/metacoma/freeplane_plugin_grpc/tree/main/misc/chatgpt

**Reply by metacoma** [reply]
hey @euu2021 
take a look https://github.com/metacoma/text2mindmap_llm

**Reply by euu2021** [reply]
I see. It's interesting that it knows how to use specific Freeplane features like details and attributes. I find it exciting to think that it's possible to have an AI model trained to use all the powerful features of Freeplane, like connectors, tags, filters etc, and use those features to express information in visual meaningful ways.

**Reply by metacoma** [reply]
Yes, here is the [AI prompt](https://github.com/metacoma/text2mindmap_llm/blob/55b3621f11b5860270024f4677b003b121276119/text2mindmap.py#L64)

Before sending the prompt, I appended the clipboard content to the end.


As you can see, the AI returns a valid JSON document in a format that is accepted by the [mindmapFromJSON](https://github.com/metacoma/freeplane_plugin_grpc/blob/b14ff10c61ef3c2e991e4b811645acef205af24f/src/main/java/org/freeplane/plugin/grpc/GrpcRegistration.java#L431) function

In the near future, I will improve [freeplane_grpc_plugin](https://github.com/metacoma/freeplane_grpc_plugin) to create nodes with tags. If you have any ideas for a better AI prompt or JSON structure, please feel free to share

**Comment by metacoma** [comment]
Hey folks, I've released a new version! (tested on OSX and Linux ubuntu)

This version adds support for relationships between nodes and tags.

**https://github.com/metacoma/text2mindmap_llm**

Screenshot of an AI-generated mind map from a random Wikipedia article.

![image](https://github.com/user-attachments/assets/72cb0e2c-ed3d-4781-8090-60bc5660789f)
